File: complete-backend/src/main/java/com/yuwan/completebackend/model/entity/Alert.java
package com.yuwan.completebackend.model.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.util.Date;

/**
 * 系统预警记录实体类
 * 对应数据库表 alert_info
 * <p>
 * 由定时任务自动生成，记录系统运行过程中检测到的各类预警事件，
 * 供系统管理员和企业负责人查看与处理。
 * </p>
 *
 * @author 系统架构师
 * @version 1.0
 * @since 2026-03-14
 */
@Data
@TableName("alert_info")
@Schema(description = "系统预警记录实体")
public class Alert implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @TableId(value = "alert_id", type = IdType.ASSIGN_ID)
    @Schema(description = "预警ID")
    private String alertId;

    @Schema(description = "预警类型（枚举值：STUDENT_NOT_SELECTED/TOPIC_NO_APPLICANT/REVIEW_BACKLOG/PHASE_DEADLINE_NEAR/SELECTION_RATE_LOW）")
    private String alertType;

    @Schema(description = "预警级别（1-提示 2-警告 3-严重）")
    private Integer alertLevel;

    @Schema(description = "预警标题")
    private String alertTitle;

    @Schema(description = "预警详情内容")
    private String alertContent;

    @Schema(description = "关联对象ID（课题ID / 学生ID / 阶段代码）")
    private String targetId;

    @Schema(description = "关联对象类型（TOPIC / STUDENT / PHASE）")
    private String targetType;

    @Schema(description = "是否已读（0-未读 1-已读）")
    private Integer isRead;

    @Schema(description = "是否已处理（0-未处理 1-已处理）")
    private Integer isResolved;

    @Schema(description = "处理时间")
    private Date resolvedAt;

    @TableLogic
    @Schema(description = "逻辑删除（0-未删除 1-已删除）")
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private Date createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private Date updateTime;
}

File: complete-frontend/src/types/alert.ts
/**
 * 系统预警相关类型定义
 * @description 智能预警中心所需的数据类型定义
 * @author 系统架构师
 * @date 2026-03-14
 */

/** 预警类型枚举 */
export enum AlertType {
  STUDENT_NOT_SELECTED = 'STUDENT_NOT_SELECTED',
  TOPIC_NO_APPLICANT   = 'TOPIC_NO_APPLICANT',
  REVIEW_BACKLOG       = 'REVIEW_BACKLOG',
  PHASE_DEADLINE_NEAR  = 'PHASE_DEADLINE_NEAR',
  SELECTION_RATE_LOW   = 'SELECTION_RATE_LOW'
}

/** 预警类型中文映射 */
export const AlertTypeLabel: Record<string, string> = {
  STUDENT_NOT_SELECTED: '学生未选报课题',
  TOPIC_NO_APPLICANT:   '课题无人选报',
  REVIEW_BACKLOG:       '课题审查积压',
  PHASE_DEADLINE_NEAR:  '阶段截止临近',
  SELECTION_RATE_LOW:   '选报率偏低'
}

/** 预警级别标签颜色映射（Ant Design Vue Tag 颜色） */
export const AlertLevelColor: Record<number, string> = {
  1: 'blue',
  2: 'orange',
  3: 'red'
}

/** 预警级别描述映射 */
export const AlertLevelLabel: Record<number, string> = {
  1: '提示',
  2: '警告',
  3: '严重'
}

/** 预警记录 VO */
export interface AlertVO {
  /** 预警ID */
  alertId: string
  /** 预警类型代码 */
  alertType: string
  /** 预警类型描述 */
  alertTypeDesc: string
  /** 预警级别（1-提示 2-警告 3-严重） */
  alertLevel: number
  /** 预警级别描述 */
  alertLevelDesc: string
  /** 预警标题 */
  alertTitle: string
  /** 预警详情 */
  alertContent: string
  /** 关联对象ID */
  targetId?: string
  /** 关联对象类型 */
  targetType?: string
  /** 是否已读（0-未读 1-已读） */
  isRead: number
  /** 是否已处理（0-未处理 1-已处理） */
  isResolved: number
  /** 处理时间 */
  resolvedAt?: string
  /** 创建时间 */
  createTime: string
  /** 更新时间 */
  updateTime?: string
}

/** 预警查询参数 */
export interface AlertQueryParams {
  alertType?: string
  alertLevel?: number
  isRead?: number
  isResolved?: number
  page?: number
  size?: number
}
